### DISCUSSION TITLE
Using PUT to Create a Resource

### DISCUSSION THREAD

**Discussion by PaulMansour** [bigskysoftware/htmx#3013] [Q&A] [answered]
Is there any (straight forward) way to use PUT to create a user-named resource?

In all of the htmx examples and books I have seen so far, generally POST is used, I assume because the URL for the PUT must be constructed dynamically at the last minute (in my case  from the current URL and some user input.) 

I think I am committed to having separate hypermedia and data APIs, so it won't be a big deal to use POST for the former and PUT for the latter, but I would prefer to use PUT and would like to make sure I'm not missing anything.

**Comment by Telroshan** [comment, marked as answer]
Hey, the doc indeed uses GET/POST, but that's simply because they are the most common & straightforward methods to showcase the library. It doesn't matter whether you use POST, PUT or PATCH, htmx will work the same.

The only thing you need to be aware of regarding methods is that, by default, GET and DELETE will have the parameters sent as URL encoded params, while POST/PUT/PATCH will have the values in the request body. This can be configured btw (see [`htmx.config.methodsThatUseUrlParams`](https://htmx.org/reference/#config))

As for your usecase, htmx indeed doesn't provide any built-in way to handle dynamic URLs. The URLs in `hx-post`, `hx-put` etc. are parsed once when htmx processes the element, but not evaluated for each request. If you really need to have the user-entered value in the URL and it cannot be a simple request parameter, then you might want to look into the [`path-params` community extension](https://github.com/bigskysoftware/htmx-extensions/blob/main/src/path-params/README.md), or roll your own relying on htmx events

Hope this helps

**Reply by PaulMansour** [reply]
Thank you, very helpful.

The path-params extension worked perfectly.
